src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -345,85 +346,19 @@ default MemorySegment allocate(long byteSize) {
     MemorySegment allocate(long byteSize, long byteAlignment);
 
     /**
-     * Creates an unbounded arena-based allocator used to allocate native memory segments.
-     * The returned allocator features a predefined block size and maximum arena size, and the segments it allocates
-     * are associated with the provided memory session. Equivalent to the following code:
-     * {@snippet lang=java :
-     * SegmentAllocator.newNativeArena(Long.MAX_VALUE, predefinedBlockSize, session);
-     * }
-     *
-     * @param session the memory session associated with the segments allocated by the arena-based allocator.
-     * @return a new unbounded arena-based allocator
-     * @throws IllegalStateException if {@code session} is not {@linkplain MemorySession#isAlive() alive}.
-     * @throws WrongThreadException if this method is called from a thread other than the thread
-     * {@linkplain MemorySession#ownerThread() owning} {@code session}.
-     */
-    static SegmentAllocator newNativeArena(MemorySession session) {
-        return newNativeArena(Long.MAX_VALUE, ArenaAllocator.DEFAULT_BLOCK_SIZE, session);
-    }
-
-    /**
-     * Creates an arena-based allocator used to allocate native memory segments.
-     * The returned allocator features a block size set to the specified arena size, and the native segments
-     * it allocates are associated with the provided memory session. Equivalent to the following code:
-     * {@snippet lang=java :
-     * SegmentAllocator.newNativeArena(arenaSize, arenaSize, session);
-     * }
-     *
-     * @param arenaSize the size (in bytes) of the allocation arena.
-     * @param session the memory session associated with the segments allocated by the arena-based allocator.
-     * @return a new unbounded arena-based allocator
-     * @throws IllegalArgumentException if {@code arenaSize <= 0}.
-     * @throws IllegalStateException if {@code session} is not {@linkplain MemorySession#isAlive() alive}.
-     * @throws WrongThreadException if this method is called from a thread other than the thread
-     * {@linkplain MemorySession#ownerThread() owning} {@code session}.
-     */
-    static SegmentAllocator newNativeArena(long arenaSize, MemorySession session) {
-        return newNativeArena(arenaSize, arenaSize, session);
-    }
-
-    /**
-     * Creates an arena-based allocator used to allocate native segments. The returned allocator features
-     * the given block size {@code B} and the given arena size {@code A}, and the native segments
-     * it allocates are associated with the provided memory session.
+     * Returns a segment allocator which responds to allocation requests by returning consecutive slices
```

Comment:
Note: newNativeArena is now gone, and replaced with a more general "slicing allocator". The reason I did this was two-fold: the arena factory seemed to be fairly specific, and not very useful to build upon (slicing allocator greatly improve on that aspect). The second problem is the name: now that we have a toplevel Arena, I think having an arena allocator factory here would be too confusing. I don't think the loss in functionality is significant (newNativeArena can be trivially implemented in terms of a slicing allocator), but if we really need it, we could always add another Arena subclass (e.g. SlicingArena which does more or less what the old allocator did.